// react-with-redux/stories.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.firstStory = exports.displayName = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _ = require('./');

var _2 = _interopRequireDefault(_);

var _redux = require('redux');

var _reactRedux = require('react-redux');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var reducer = function reducer(state, action) {
  console.log(action);
  return { things: [1, 2, 3] };
};
var store = (0, _redux.createStore)(reducer);

// displayName will be the title of the story
var displayName = exports.displayName = 'WithRedux';

var firstStory = exports.firstStory = {
  title: 'WithRedux First Story',
  story: function story(_ref) {
    var action = _ref.action;
    return function () {
      return _react2.default.createElement(
        _reactRedux.Provider,
        { store: store },
        _react2.default.createElement(
          _2.default,
          null,
          function (state, dispatch) {
            return state.things.map(function (t) {
              return _react2.default.createElement(
                'div',
                null,
                _react2.default.createElement(
                  'a',
                  { onClick: function onClick() {
                      return dispatch({ type: t });
                    } },
                  ' ',
                  t,
                  ' '
                )
              );
            });
          }
        )
      );
    };
  }
};
